project-addons/flask_middleware_connector/events/invoice_events.py

```diff
@@ -71,18 +70,22 @@ def delay_write_invoice(session, model_name, record_id, vals):
     up_fields = ["number", "client_ref", "date_invoice", "state", "partner_id",
                  "date_due", "subtotal_wt_rect", "subtotal_wt_rect"]
 
-    if invoice.partner_id and invoice.partner_id.web:
-        if vals.get('state', False) == 'open':
-            export_invoice.delay(session, model_name, record_id)
-        elif vals.get('state', False) == 'paid':
-            update_invoice.delay(session, model_name, record_id)
-        elif vals.get('state', False) == 'cancel':
-            unlink_invoice(session, model_name, record_id)
+    if invoice.partner_id and invoice.commercial_partner_id.web:
+        job = session.env['queue.job'].search(['&', ('func_string', 'like', '%' + str(invoice.id) + '%'), ('model_name', '=', model_name)])
```

Comment:
Tiene que añadirle un orden y un limit, sino no sabes cual te va a devolver
añádele: order='date_created desc' y limit=1
Además tiene que obviar los trabajos del SII. Un domain mejor sería:
[('func_string', 'not like' '%confirm_one_invoice%'),('func_string', 'like', '%, ' + str(invoice.id) + ')%'),
 ('model_name', '=', model_name)]

Además así evitamos que entre con ids que contengan dentro de sus número el id de factura por el que buscamos.